File: Sorting/Selection/SelectionSort.java
package Sorting;

public class SelectionSort {
    public static void main(String[] args) {
        int arr[] = {-3,6,-5,2,16,1};

        selectionSort(arr);

        for(int i=0;i<arr.length;i++)
            System.out.print(arr[i]+ " ");
    }

    static void selectionSort(int arr[]) {
        int minIndex, temp;

        for(int i=0;i<arr.length;i++) {
            minIndex= i;

            for(int j=i+1;j<arr.length;j++) {
                if(arr[j] < arr[minIndex])
                    minIndex = j;
            }

            temp = arr[i];
            arr[i] = arr[minIndex];
            arr[minIndex] = temp;
        }
    }
}

File: Sorting/Selection/SelectionSort.cpp
#include<iostream>

using namespace std;

void selectionSort(int arr[], int length) {
    int minIndex, temp;

    for(int i=0; i<length; i++) {
        minIndex= i;

        for(int j=i+1;j<length;j++) {
            if(arr[j] < arr[minIndex])
                minIndex = j;
        }

        temp = arr[i];
        arr[i] = arr[minIndex];
        arr[minIndex] = temp;
    }
}

int main() {
    int arr[] = {-3,6,-5,2,16,1};
    int length = 6;

    selectionSort(arr, length);

    for(int i=0;i<length;i++)
        cout << arr[i] << " ";

    return 0;
}